spiceypy/spiceypy.py

```diff
@@ -77,16 +77,14 @@ def check_for_spice_error(f: Optional[Callable]) -> None:
     :raise stypes.SpiceyError:
     """
     if failed():
-        errorparts = {
-            "tkvsn": tkvrsn("TOOLKIT").replace("CSPICE_", ""),
-            "short": getmsg("SHORT", 26),
-            "explain": getmsg("EXPLAIN", 100).strip(),
-            "long": getmsg("LONG", 321).strip(),
-            "traceback": qcktrc(200),
-        }
-        msg = stypes.errorformat.format(**errorparts)
+        short = getmsg("SHORT", 26)
+        explain = getmsg("EXPLAIN", 100).strip()
+        long = getmsg("LONG", 321).strip()
+        traceback = qcktrc(200)
         reset()
-        raise stypes.SpiceyError(msg)
+        raise stypes.dynamically_instantiate_spiceyerror(
```

Comment:
Ah, so this is cool. You've gone from making the error just based on `msg` to also using the other parts of the error metadata, including traceback.